garak/resources/beast/beast_attack.py

```diff
@@ -0,0 +1,403 @@
+        beams = [candidates[j] for j in sorted_scores[:k1]]
+
+        best_candidate = candidates[sorted_scores[0]]
+        candidate_score = scores[sorted_scores[0]]
+
+        if candidate_score < best_score:
+            best_score = candidate_score
+            best_suffix = best_candidate
+
+        if stop_early:
+            success = _evaluate(generator, prompt, best_candidate)
+            if success:
+                break
+
+    return best_suffix, best_score
+
+
+def _attack(
+    generator: Generator,
+    prompts: list[str],
+    responses: Optional[list[str]] = None,
+    k1: int = 15,
+    k2: int = 15,
+    trials: int = 1,
+    suffix_len: int = 40,
+    target: Optional[str] = "",
+    stop_early: bool = False,
+) -> list[str]:
+    """
+
+    Args:
+        generator: target generator to attack
+        prompts: Input prompt
+        responses: Responses for input prompts
+        k1: Number of candidates in beam
+        k2: Number of candidates to evaluate
+        trials: Number of generations to run for the attack
+        suffix_len: Number of adversarial tokens to generate
+        target: Target output string
+        stop_early: Whether to stop if a successful jailbreak is found
+
+    Returns:
+        prompt_tokens: Adversarial prompt tokens
+        scores: Score for attack objective per item in prompt_tokens
+    """
+    suffixes = list()
+    if responses is None:
+        responses = ["" for _ in range(len(prompts))]
+    for prompt, response in tqdm(
+        zip(prompts, responses), total=len(prompts), leave=False, position=0
+    ):
+        best_candidate = []
+        if trials > 1:
+            pbar = tqdm(total=trials, leave=False)
+
+        for _ in range(trials):
+            best_candidate, score = _get_best_candidate(
+                generator,
+                prompt,
+                response,
+                k1,
+                k2,
+                suffix_len,
+                best_candidate,
+                stop_early,
+            )
+
+            if target:
+                result, response = _evaluate_target(
+                    generator, prompt, best_candidate, target
+                )
+            else:
+                result, response = _evaluate(generator, prompt, best_candidate)
+
+            if result:
+                jailbreak_str = generator.tokenizer.decode(best_candidate)
+                logging.info("BEAST found a likely successful jailbreak")
+                suffixes.append(jailbreak_str)
+
+            if trials > 1:
+                pbar.update(1)
+
+    return suffixes
+
+
+def run_beast(
+    target_generator: garak.generators.Generator = None,
+    prompts: Optional[list[str]] = None,
+    responses: Optional[list[str]] = None,
+    k1: int = 15,
+    k2: int = 15,
+    trials: int = 1,
+    suffix_len: int = 40,
+    data_size: int = 20,
+    target: Optional[str] = "",
+    outfile: str = beast_resource_data / "suffixes.txt",
```

Comment:
This default location is inside the installed path, I would suggest this should be either in the OS temp directory or a user's profile/configuration directory.

This is likely a reasonable choice to get this PR staged and landed, tagging #479 to note location that may need update.

Reply:
one temp solution is something like `garak._config.reporting.report_dir / "garak." + garak._config.transient.run_id + ".beast_suffixes.txt"`